packages/holocron-dev-server/__tests__/src/webpack/plugins/holocron-webpack-plugin.spec.js

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2021 American Express Travel Related Services Company, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { getModuleFromFilePath } from '../../../../src/utils/helpers';
+
+import HolocronWebpackPlugin from '../../../../src/webpack/plugins/holocron-webpack-plugin';
+import { getWebpackVersion } from '../../../../src/webpack/helpers';
+
+jest.mock('../../../../src/webpack/helpers', () => ({
+  getWebpackVersion: jest.fn(() => 5),
+}));
+
+jest.mock('../../../../src/utils/helpers', () => {
+  const originalModule = jest.requireActual('../../../../src/utils/helpers');
+
+  return {
+    ...originalModule,
+    getModuleFromFilePath: jest.fn(() => true),
+  };
+});
+beforeEach(() => {
+  jest.clearAllMocks();
+});
+
+describe('HolocronWebpackPlugin', () => {
+  test('instantiates without error', () => {
+    expect(() => new HolocronWebpackPlugin()).not.toThrow();
+  });
+
+  test('plugin is applied to compilation and taps into loader webpack hook for version 5', () => {
+    const instance = new HolocronWebpackPlugin();
```

Comment:
not critical but i would try to avoid generics like instance, holocronWebpackPlugin or if really wanted holocronWebpackPluginInstance